apps/rehabstod/src/pages/LUCertificates/utils/luCertificatesTableValueFormatter.tsx

```diff
@@ -0,0 +1,48 @@
+import { LUCertificate } from '../../../schemas/luCertificatesSchema'
+import { LUCertificatesColumn } from '../../../store/slices/luCertificatesTableColumns.slice'
+import { TableCell } from '../../../components/Table/tableBody/TableCell'
+import { DiagnosisInfo } from '../../../components/Diagnosis/DiagnosisInfo'
+import { DiagnosisDescription } from '../../../components/Diagnosis/DiagnosisDescription'
+import { getUnansweredCommunicationFormat } from '../../../components/UnansweredCommunication/utils/getUnansweredCommunicationFormat'
+
+export const getLUCertificatesTableValue = (column: string, data: LUCertificate) => {
+  switch (column) {
+    case LUCertificatesColumn.Personnummer:
+      return data.patient.id
+    case LUCertificatesColumn.Ålder:
+      return data.patient.alder
+    case LUCertificatesColumn.Kön:
+      return data.patient.kon === 'F' ? 'Kvinna' : 'Man'
+    case LUCertificatesColumn.Namn:
+      return data.patient.namn
+    case LUCertificatesColumn.Intyg:
+      return data.certificateType
+    case LUCertificatesColumn.Signeringsdatum:
+      return data.signingTimeStamp
+    case LUCertificatesColumn.Läkare:
+      return data.doctor.namn
+    case LUCertificatesColumn.Ärenden:
+      return data.unAnsweredComplement + data.unAnsweredOther
+    case LUCertificatesColumn.Diagnos:
+      return data.diagnosis ? data.diagnosis.kod : 'Okänt'
+    default:
+      return undefined
+  }
+}
+
+export const getLUCertificatesTableCell = (column: string, data: LUCertificate) => {
+  switch (column) {
+    case LUCertificatesColumn.Diagnos:
+      return (
+        <TableCell description={<DiagnosisDescription diagnos={data.diagnosis} biDiagnoser={data.biDiagnoses} />}>
+          <DiagnosisInfo biDiagnoser={data.biDiagnoses} diagnos={data.diagnosis} />
+        </TableCell>
+      )
+    case LUCertificatesColumn.Signeringsdatum:
+      return <TableCell>{(getLUCertificatesTableValue(column, data) as string).split('T')[0]}</TableCell>
+    case LUCertificatesColumn.Ärenden:
+      return <TableCell>{getUnansweredCommunicationFormat(data.unAnsweredComplement, data.unAnsweredOther)}</TableCell>
+    default:
+      return <TableCell>{getLUCertificatesTableValue(column, data)}</TableCell>
+  }
+}
```

Comment:
This is returning `JSX.Element`.  It should be moved and made into a component.

Reply:
I don't agree with this I think it will make the TableBody less understandable

Reply:
With this solution, the array of element is not properly assigned key and React can't handle re-rendering properly.

Reply:
If I change it to a component how can I write it in a way that works for several tables? Also, I don't get any errors in the console that there should be problems with the re-rendering of the table

Reply:
Added a potential solution for this